rakam-postgresql/src/main/java/org/rakam/postgresql/analysis/FastGenericFunnelQueryExecutor.java

```diff
@@ -46,7 +47,7 @@ public FastGenericFunnelQueryExecutor(QueryExecutorService executor, ProjectConf
     }
 
     @Override
-    public QueryExecution query(String project, List<FunnelStep> steps, Optional<String> dimension, LocalDate startDate, LocalDate endDate, Optional<FunnelWindow> window, ZoneId timezone, Optional<List<String>> connectors, FunnelType funnelType)
+    public QueryExecution query(String project, List<FunnelStep> steps, Optional<String> dimension, Optional<String> segment, LocalDate startDate, LocalDate endDate, Optional<FunnelWindow> window, ZoneId timezone, Optional<List<String>> connectors, FunnelType funnelType)
     {
         if(dimension.map(v -> projectConfig.getTimeColumn().equals(v) || projectConfig.getUserColumn().equals(v)).orElse(false)) {
```

Comment:
It would be nice to add check for the following cases:
If the dimension is a `TIMESTAMP` column but segment is not enabled. (The TIMESTAMP columns will have high cardinality so the result won't be meaningful)
If the dimension is not a 'TIMESTAMP' column but segment is enabled. (The is to segment available for columns that are not `TIMESTAMP`)

Reply:
Added these checks